### PR TITLE
Spelling

### LINKED ISSUE #8279: firebase-js-sdk has typos
### Operating System

N/A

### Browser Version

N/A

### Firebase SDK Version

master

### Firebase SDK Product:

Installations

### Describe your project's tooling

The [check-spelling action](https://github.com/marketplace/actions/check-spelling) identifies misspellings and other errors using words.

### Describe the problem

The misspellings have been reported at https://github.com/jsoref/firebase-js-sdk/actions/runs/9310397584#summary-25627717291

### Steps and code to reproduce issue

1. Fork this repository
2. Add https://github.com/jsoref/firebase-js-sdk/commit/97e22c4f2ef056ec3c8d5095300453a62f682842
3. Push to the fork

### PR DESCRIPTION
**Pull request by jsoref**
### Discussion

Fixes #8279

### Testing

  * Make sure all existing tests in the repository pass after your change. -- They do (although, many of them required fixes in #8275 for me to validate)
  * If you fixed a bug or added a feature, add a new test to cover your code. -- This can be done by applying a variant of https://github.com/jsoref/firebase-js-sdk/commit/spell-check, but I'd prefer to leave that outside the scope of this PR

### API Changes

I believe there's one API change included:
https://github.com/jsoref/firebase-js-sdk/commit/10eacaec7a576fc1dea6110af813292c362586e8

I've included it in the hopes that it isn't an API and, thus, can be changed. If it's an API and can't be changed, it's trivial for me to drop it (just as I'm more than happy to drop any other changes).

### CODE REVIEW

File: docs-devsite/database.datasnapshot.md [hunk 1]
```diff
@@ -38,7 +38,7 @@ export declare class DataSnapshot
...
+|  [forEach(action)](./database.datasnapshot.md#datasnapshotforeach) |  | Enumerates the top-level children in the <code>IteratedDataSnapshot</code>.<!-- -->Because of the way JavaScript objects work, the ordering of data in the JavaScript object returned by <code>val()</code> is neither guaranteed to match the ordering on the server nor the ordering of <code>onChildAdded()</code> events. That is where <code>forEach()</code> comes in handy. It guarantees the children of a <code>DataSnapshot</code> will be iterated in their query order.<!-- -->If no explicit <code>orderBy*()</code> method is used, results are returned ordered by key (unless priorities are used, in which case, results are returned by priority). |
```
Comment (jsoref) [inline docs-devsite/database.datasnapshot.md:41]: I'm happy to drop any of my proposed changes.

neither/nor is the generally recommended way to write this, and as this is documentation, using it seems reasonable.

File: .github/ISSUE_TEMPLATE/bug_report_v2.yaml [hunk 2]
```diff
@@ -22,7 +22,7 @@ body:
     attributes:
       value: |
         *[READ THIS]:* to evaluate if you are in the right place?
-        - For issues or feature requests related to __the code in this repository__, file a Github issue.
+        - For issues or feature requests related to __the code in this repository__, file a GitHub issue.
```
Comment (jsoref) [inline .github/ISSUE_TEMPLATE/bug_report_v2.yaml:25]: brand

File: docs-devsite/firestore_.md [hunk 3]
```diff
@@ -2598,7 +2598,7 @@ export declare type OrderByDirection = 'desc' | 'asc';
 
 ## PartialWithFieldValue
 
-Similar to Typescript's `Partial<T>`<!-- -->, but allows nested fields to be omitted and FieldValues to be passed in as property values.
+Similar to TypeScript's `Partial<T>`<!-- -->, but allows nested fields to be omitted and FieldValues to be passed in as property values.
```
Comment (jsoref) [inline docs-devsite/firestore_.md:2601]: brand

File: packages/app-check/src/api.test.ts [hunk 4]
```diff
@@ -10,7 +10,7 @@
  *
  * Unless required by applicable law or agreed to in writing, software
  * distributed under the License is distributed on an "AS IS" BASIS,
- * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or ied.
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
```
Comment (jsoref) [inline packages/app-check/src/api.test.ts:13]: generally, lawyers get annoyed by changes to licenses...

A peer suggested that someone did a global search and replace `s/impl//`. -- I believe they're correct.

File: packages/app-compat/src/public-types.ts [hunk 5]
```diff
@@ -85,7 +85,7 @@ export interface FirebaseNamespace {
      *
      * DO NOT call this constuctor directly (use firebase.app() instead).
      */
-    App: FirebaseAppContructor;
+    App: FirebaseAppConstructor;
```
Comment (jsoref) [inline packages/app-compat/src/public-types.ts:88]: As noted in the initial PR, this is the only change that might be an api change. Given the comment here, I'd hope it isn't particularly problematic. Anyone upgrading would just need to fix the spelling of the type.

File: packages/auth/src/api/authentication/mfa.test.ts [hunk 6]
```diff
@@ -34,7 +34,7 @@ describe('api/authentication/startSignInPhoneMfa', () => {
     mfaPendingCredential: 'my-creds',
     mfaEnrollmentId: 'my-enrollment-id',
     phoneSignInInfo: {
-      recaptchaToken: 'catpcha-token'
+      recaptchaToken: 'captcha-token'
```
Comment (jsoref) [inline packages/auth/src/api/authentication/mfa.test.ts:37]: ??

File: packages/auth/src/api/index.test.ts [hunk 7]
```diff
@@ -367,11 +367,11 @@ describe('api/_performApiRequest', () => {
     });
   });
 
-  context('edgcase error mapping', () => {
+  context('edge cage error mapping', () => {
```
Comment (jsoref) [inline packages/auth/src/api/index.test.ts:370]: it's possible `edgcase` is a meaningful thing, but I'm assuming it isn't

File: packages/auth/src/platform_browser/recaptcha/recaptcha_loader.ts [hunk 8]
```diff
@@ -77,7 +77,7 @@ export class ReCaptchaLoaderImpl implements ReCaptchaLoader {
           return;
         }
 
-        // Wrap the greptcha render function so that we know if the developer has
+        // Wrap the recaptcha render function so that we know if the developer has
```
Comment (jsoref) [inline packages/auth/src/platform_browser/recaptcha/recaptcha_loader.ts:80]: It's possible this should be `grecaptcha`, but based on the code below using `recaptcha`, I guessed this way...

File: packages/auth/test/integration/webdriver/redirect.test.ts [hunk 9]
```diff
@@ -232,7 +232,7 @@ browserDescribe('WebDriver redirect IdP test', driver => {
       'user@test.test'
     );
 
-    // Link using pre-poulated user
+    // Link using pre-populated user
```
Comment (jsoref) [inline packages/auth/test/integration/webdriver/redirect.test.ts:235]: If I make updates to this again, I'm likely to change it to `prepopulated` (dropping the `-`)

File: packages/auth-compat/demo/public/manifest.json [hunk 10]
```diff
@@ -1,5 +1,5 @@
 {
-  "name": "Firebse Auth Test App",
+  "name": "Firebase Auth Test App",
```
Comment (jsoref) [inline packages/auth-compat/demo/public/manifest.json:2]: It's possible this is intentional, but it didn't seem to be...

File: packages/auth-compat/CHANGELOG.md [hunk 11]
```diff
@@ -332,7 +332,7 @@
 
 ### Patch Changes
 
-- [`63ac2ed28`](https://github.com/firebase/firebase-js-sdk/commit/63ac2ed28f237950290a7af2dcdcf1518ddaee4b) - Fix auth scheme reconition in capacitor env
+- [`63ac2ed28`](https://github.com/firebase/firebase-js-sdk/commit/63ac2ed28f237950290a7af2dcdcf1518ddaee4b) - Fix auth scheme recognition in capacitor env
```
Comment (jsoref) [inline packages/auth-compat/CHANGELOG.md:335]: Some projects don't like changing changelogs. Happy to drop if this is a problem

File: packages/database/src/api/Database.ts [hunk 12]
```diff
@@ -303,10 +303,9 @@ export function forceLongPolling() {
 }
 
 /**
- * Returns the instance of the Realtime Database SDK that is associated
- * with the provided {@link @firebase/app#FirebaseApp}. Initializes a new instance with
- * with default settings if no instance exists or if the existing instance uses
- * a custom database URL.
+ * Returns the instance of the Realtime Database SDK that is associated with the provided
+ * {@link @firebase/app#FirebaseApp}. Initializes a new instance with default settings if
+ * no instance exists or if the existing instance uses a custom database URL.
```
Comment (jsoref) [inline packages/database/src/api/Database.ts:308]: This was frustrating as the old code ended one line with `with` and started the next line (after the `*`) with `with`.

My tooling easily caught the rendered form, but it took me a while to find this case (it happens occasionally, but not enough for me to try to teach my tooling to look for such things, especially since there's often intervening punctuation, as here).

The rewrapping is an attempt to match the `@param app` line lenght below -- especially given that nothing seems to care.

File: packages/database/test/helpers/syncPointSpec.json [hunk 13]
```diff
@@ -3729,7 +3729,7 @@
         "data": {
           "4": { "age": 41, "highscore": 400, "name": "old mama"},
           "5": { "age": 18, "highscore": 1200, "name": "young mama"},
-          "6": { "age": 20, "highscore": 1003, "name": "micheal blub"},
+          "6": { "age": 20, "highscore": 1003, "name": "michael blub"},
```
Comment (jsoref) [inline packages/database/test/helpers/syncPointSpec.json:3732]: I can't really tell what's going on here. It probably doesn't matter at all, but it's pretty odd to write `micheal` and google search for "micheal blub" (when I beg it to let me search for it) gives me the cute "go fishing" animation.

Happy to drop.

File: packages/firebase/compat/rollup.config.js [hunk 14]
```diff
@@ -19,7 +19,7 @@ import { resolve } from 'path';
 import resolveModule from '@rollup/plugin-node-resolve';
 import commonjs from '@rollup/plugin-commonjs';
 import sourcemaps from 'rollup-plugin-sourcemaps';
-import rollupTypescriptPlugin from 'rollup-plugin-typescript2';
+import rollupTypeScriptPlugin from 'rollup-plugin-typescript2';
```
Comment (jsoref) [inline packages/firebase/compat/rollup.config.js:22]: Ooops. I tried to find all the interesting things.

I'll drop this.

File: packages/firestore/src/local/memory_remote_document_cache.ts [hunk 15]
```diff
@@ -205,7 +205,7 @@ class MemoryRemoteDocumentCacheImpl implements MemoryRemoteDocumentCache {
     transaction: PersistenceTransaction,
     collectionGroup: string,
     offset: IndexOffset,
-    limti: number
+    limit: number
```
Comment (jsoref) [inline packages/firestore/src/local/memory_remote_document_cache.ts:208]: ??

File: packages/firestore/test/unit/remote/bloom_filter.test.ts [hunk 16]
```diff
@@ -186,13 +186,13 @@ describe('BloomFilter', () => {
         TEST_DATA.count5000Rate1TestResult
       );
     });
-    it('mightContain result for 5000 documenta with 0.01 false positive rate', () => {
+    it('mightContain result for 5000 documents with 0.01 false positive rate', () => {
```
Comment (jsoref) [inline packages/firestore/test/unit/remote/bloom_filter.test.ts:189]: ??

File: packages/performance/src/services/perf_logger.ts [hunk 17]
```diff
@@ -85,7 +85,7 @@ interface TraceMetric {
   custom_attributes?: { [key: string]: string };
 }
 
-/* eslint-enble camelcase */
+/* eslint-disable camelcase */
```
Comment (jsoref) [inline packages/performance/src/services/perf_logger.ts:88]: notable. The obvious change `eslint-enable` results in a lot of failures

File: scripts/docgen-compat/theme/assets/css/main.css [hunk 18]
```diff
@@ -518,7 +518,7 @@ ul.tsd-descriptions.active > .tsd-description.fade-out { -webkit-animation: fade
 ul.tsd-descriptions h4, ul.tsd-descriptions .tsd-index-panel h3, .tsd-index-panel ul.tsd-descriptions h3 { font-size: 16px; margin: 1em 0 0.5em 0; }
 
 ul.tsd-parameters, ul.tsd-type-parameters { list-style: square; margin: 0; padding-left: 20px; }
-ul.tsd-parameters > li.tsd-parameter-siganture, ul.tsd-type-parameters > li.tsd-parameter-siganture { list-style: none; margin-left: -20px; }
+ul.tsd-parameters > li.tsd-parameter-signature, ul.tsd-type-parameters > li.tsd-parameter-signature { list-style: none; margin-left: -20px; }
```
Comment (jsoref) [inline scripts/docgen-compat/theme/assets/css/main.css:521]: notable

File: packages/firestore/test/integration/api/snapshot_listener_source.test.ts
Comment (jsoref) [inline packages/firestore/test/integration/api/snapshot_listener_source.test.ts:1]:
```diff
-snasphot_listener_source.test.ts
+snapshot_listener_source.test.ts
```

File: repo-scripts/prune-dts/tests/resolves-generics-through-inheritance.input.d.ts
Comment (jsoref) [inline repo-scripts/prune-dts/tests/resolves-generics-through-inheritance.input.d.ts:1]:
```diff
-inheritence.input.d.ts
+inheritance.input.d.ts
```

File: docs-devsite/database.datasnapshot.md (hunk 1 above)
Comment (egilmorez) [inline docs-devsite/database.datasnapshot.md:41]: I think the placement of this is off; "neither" should be placed right before the first instance of "the ordering" in order to clarify which things we are contrasting. As it stands, it would be more logical to say something like "neither guaranteed nor recommended," which is not what we mean.

Here and below.

File: docs-devsite/vertexai-preview.chatsession.md [hunk 19]
```diff
@@ -36,7 +36,7 @@ export declare class ChatSession
 
 |  Method | Modifiers | Description |
 |  --- | --- | --- |
-|  [getHistory()](./vertexai-preview.chatsession.md#chatsessiongethistory) |  | Gets the chat history so far. Blocked prompts are not added to history. Blocked candidates are not added to history, nor are the prompts that generated them. |
+|  [getHistory()](./vertexai-preview.chatsession.md#chatsessiongethistory) |  | Gets the chat history so far. Blocked prompts are not added to history. Blocked candidates are neither added to history nor are the prompts that generated them. |
```
Comment (egilmorez) [inline docs-devsite/vertexai-preview.chatsession.md:39]: This is another poor usage IMO.  I'd stick with just "not," here and below.

If we insist on using neither, the correct way would be "Neither blocked candidates nor the prompts that generated them are . . ."

File: packages/auth/src/api/index.test.ts (hunk 7 above)
Comment (egilmorez) [reply packages/auth/src/api/index.test.ts:370]: Suggest "edge case" instead of "cage."

File: packages/database/src/api/Reference_impl.ts [hunk 20]
```diff
@@ -379,7 +379,7 @@ export class DataSnapshot {
    * Enumerates the top-level children in the `IteratedDataSnapshot`.
    *
    * Because of the way JavaScript objects work, the ordering of data in the
-   * JavaScript object returned by `val()` is not guaranteed to match the
+   * JavaScript object returned by `val()` is neither guaranteed to match the
```
Comment (egilmorez) [inline packages/database/src/api/Reference_impl.ts:382]: Same positioning problem noted above.  Place before "the ordering"

File: packages/vertexai/src/methods/chat-session.ts [hunk 21]
```diff
@@ -61,7 +61,7 @@ export class ChatSession {
 
   /**
    * Gets the chat history so far. Blocked prompts are not added to history.
-   * Blocked candidates are not added to history, nor are the prompts that
+   * Blocked candidates are neither added to history nor are the prompts that
```
Comment (egilmorez) [inline packages/vertexai/src/methods/chat-session.ts:64]: Suggest just sticking with "not" (or move "neither" where it makes more sense).

**Review by egilmorez — COMMENTED** [review]
This is a super useful effort that I enthusiasticallyt endorse (could the same tooling be used on similar TypeScript source, like the Cloud Functions reference?), but wow –  over 200 files is a lot to review at once, for a human. Next time around, would be great to break it down into categories of some kind, like proper naming issues (Github> GitHub),  misspellings of nouns, misspellings of verbs, whatever.

Also, for some of the more sophisticated errata, we may need to tweak the algorithm to prevent the neither/nor errors that I picked out. Hope I didn't miss any similar things ( placement of "only"  often causes similar issues).

File: packages/auth/src/api/index.test.ts (hunk 7 above)
Comment (jsoref) [reply packages/auth/src/api/index.test.ts:370]: Sigh. I have no idea how I messed that one up.

File: docs-devsite/database.datasnapshot.md (hunk 1 above)
Comment (jsoref) [reply docs-devsite/database.datasnapshot.md:41]: I don't think your suggestion makes sense. I'll drop the change.

**Comment by jsoref** [comment]
The changes are all manual (more or less), and anything that isn't a simple replacement is very manual (especially things like `neither`).

Splitting is more or less possible -- it's easy for me to group commits together, or split by directory. It's fairly painful for me to split by part of speech if a word is used in more than one part of speech (and gets prohibitive as the word count grows -- it's error prone, and as a human, I make mistakes like everyone else), but it's certainly not a problem for me to peel off individual commits or group commits together into separate PRs -- I just need direction. For smaller repositories, I can split by "code" vs "comment" -- although I really don't like to do it as it isn't particularly safe -- it's much cleaner to split by directory or possibly file extension.

The only thing that's fully automated is identifying potential problems. I can certainly run the tooling on any repository if people are interested. Sometimes I rely on Google Sheets for word corrections, although of late, I've just been using Google Chrome suggestions.

File: docs-devsite/database.datasnapshot.md (hunk 1 above)
Comment (egilmorez) [reply docs-devsite/database.datasnapshot.md]: I dunno Josh, ever since I scored in the 99th percentile of the verbal GRE with a 780, I've been pretty confident about writing the English -- and that was a long time ago!  :D

To clarify, I'm not suggesting "neither guaranteed nor recommended" -- that's just an example of how to resolve the tension created by "neither guaranteed."

What I *would* suggest, is one of these:

a) "not guaranteed to match . . ."

b) "is guaranteed to match neither the ordering on the server, nor the ordering of `onChildAdded()` events."

And I absolutely DO guarantee that both of those are grammatically correct.  I'd vote for a) any day due to its economy and simplicity, but you seemed to be showing a liking for "neither" constructions.

Comment (jsoref) [reply docs-devsite/database.datasnapshot.md]: Sorry, the problem with your initial suggestion is that there were _three_ `the ordering`s in the text, and I suspect you actually wanted to point me to the _second_ one.

Here's my proposal to reduce the complexity there (I'm not include the markup or the other sentence as it doesn't matter -- I'd have to apply any changes to multiple places if they're accepted anyway):

> JavaScript objects do not guarantee that the order of data returned by val() will match either order of events from the server or from onChildAdded() events.

I'm happy to restore some words at the beginning if I'm wrong in my guess that those other words I've dropped aren't particularly essential.

That said, if you don't think that changes to this text will be fruitful, then I'd rather leave the changes dropped (as they are right now) and focus on the remainder of the PR.

Comment (egilmorez) [reply docs-devsite/database.datasnapshot.md]: Probably leaving it as/is works best.  The suggested text feels like is lacking a definite article before "order of events," so I'd stay away from that phrasing.

**Review by egilmorez — APPROVED** [review]
I think jsoref fixed or backed out content related to my neither/nor suggestions (not easy to immediately discover that due to the massive size of this PR), so I think I'm happy to approve.  Thanks!

**Comment by jsoref** [comment]
Do I need to worry about the changeset bot?

Note that https://github.com/firebase/firebase-js-sdk/actions/runs/9639724064/job/26582355096?pr=8280 is failing (I can't remember the state of this, I split most other CI failures into other PRs...)

File: packages/firestore/src/local/memory_remote_document_cache.ts (hunk 15 above)
Comment (MarkDuckworth) [reply packages/firestore/src/local/memory_remote_document_cache.ts:208]: LGTM

File: packages/firestore/test/unit/remote/bloom_filter.test.ts (hunk 16 above)
Comment (MarkDuckworth) [reply packages/firestore/test/unit/remote/bloom_filter.test.ts:189]: lgtm

File: .github/ISSUE_TEMPLATE/bug_report_v2.yaml (hunk 2 above)
Comment (MarkDuckworth) [reply .github/ISSUE_TEMPLATE/bug_report_v2.yaml:25]: LGTM

File: scripts/docgen-compat/theme/assets/css/main.css (hunk 18 above)
Comment (MarkDuckworth) [reply scripts/docgen-compat/theme/assets/css/main.css:521]: It's unclear if this CSS class name change was also made in the corresponding HTML. If not, maybe revert this?

**Review by MarkDuckworth — APPROVED** [review]
LGTM with comments. I reviewed `packages/firestore` and other top-level folders outside of `packages/`.

**Review by DellaBitta — CHANGES_REQUESTED** [review]
Hi @jsoref, thanks for the PR!

Could you add a changeset (run `yarn changeset`) and mark all packages with code changes as `patch` with the note `Fixed typos in documentation and in some internal variables and parameters.` ?

Also looks like you have a format error. Could your run the following commands:

`yarn format`
`yarn docgen:all`

You _might_ need to merge the main branch into your branch because there were some changes to the documentation script a few weeks ago.

File: packages/performance/src/services/perf_logger.ts (hunk 17 above)
Comment (jsoref) [reply packages/performance/src/services/perf_logger.ts:88]: I hope you mean `/* eslint-enable camelcase */` (for line 249).

Comment (DellaBitta) [reply packages/performance/src/services/perf_logger.ts:88]: Hah, yes. Copy pasta.

**Comment by jsoref** [comment]
@DellaBitta: I can't get yarn to cooperate
```js
jsoref@jsoref-mbp firebase-js-sdk % node --version
v20.11.0
jsoref@jsoref-mbp firebase-js-sdk % yarn
➤ YN0000: · Yarn 4.2.1
➤ YN0000: ┌ Resolution step
➤ YN0001: │ Error: closure-net@https://github.com/google/closure-net.git#commit=0412666e8f29b8ae69decb1fdc7ead635a5cf43e: Manifest not found
    at AE.find (/Users/jsoref/.yarn/releases/yarn-4.2.1.cjs:140:120021)
    at async /Users/jsoref/.yarn/releases/yarn-4.2.1.cjs:520:11979
    at async Object.xZe (/Users/jsoref/.yarn/releases/yarn-4.2.1.cjs:140:53847)
    at async I2.resolve (/Users/jsoref/.yarn/releases/yarn-4.2.1.cjs:520:11940)
    at async Dd.resolve (/Users/jsoref/.yarn/releases/yarn-4.2.1.cjs:141:1451)
    at async Dd.resolve (/Users/jsoref/.yarn/releases/yarn-4.2.1.cjs:141:1451)
    at async /Users/jsoref/.yarn/releases/yarn-4.2.1.cjs:210:7249
    at async Ky (/Users/jsoref/.yarn/releases/yarn-4.2.1.cjs:140:53910)
    at async ne (/Users/jsoref/.yarn/releases/yarn-4.2.1.cjs:210:7231)
    at async Promise.allSettled (index 1)
➤ YN0000: └ Completed in 0s 873ms
➤ YN0000: · Failed with errors in 0s 883ms
```

I've given permission to push into my branch...

**Comment by DellaBitta** [comment]
Done! Waiting for CI to execute now.

**Review by DellaBitta — APPROVED** [review]
